internal/service/common.go

```diff
@@ -11,3 +21,63 @@ func NilOutManagedObjectMetaProperties(om *api.ObjectMeta) {
 	om.CreationTimestamp = nil
 	om.DeletionTimestamp = nil
 }
+
+func validateAgainstSchema(obj []byte, objPath string) error {
+	swagger, err := api.GetSwagger()
+	if err != nil {
+		return err
+	}
+	// Skip server name validation
+	swagger.Servers = nil
+
+	url, err := url.Parse(objPath)
+	if err != nil {
+		return err
+	}
+	httpReq := &http.Request{
+		Method: "PUT",
+		URL:    url,
+		Body:   io.NopCloser(bytes.NewReader(obj)),
+		Header: http.Header{"Content-Type": []string{"application/json"}},
+	}
+
+	router, _ := gorillamux.NewRouter(swagger)
+	route, pathParams, _ := router.FindRoute(httpReq)
+
+	requestValidationInput := &openapi3filter.RequestValidationInput{
+		Request:    httpReq,
+		PathParams: pathParams,
+		Route:      route,
+	}
+	return openapi3filter.ValidateRequest(context.Background(), requestValidationInput)
```

Comment:
please ensure this and friends have a real context, it will help with graceful termination flow.

Reply:
great catch. I'm passing a real context now.